prog/download_boot_image.rb

```diff
@@ -30,9 +38,10 @@ def custom_url
       sshable.cmd("common/bin/daemonizer --clean download_#{image_name.shellescape}")
       hop_learn_storage
     when "NotStarted"
-      sshable.cmd("common/bin/daemonizer 'host/bin/download-boot-image #{image_name.shellescape} #{custom_url.shellescape}' download_#{image_name.shellescape}")
+      url = custom_url || blob_storage_client.get_presigned_url("GET", Config.ubicloud_images_bucket_name, "#{image_name}-#{vm_host.arch}.raw").to_s
+      sshable.cmd("common/bin/daemonizer 'host/bin/download-boot-image #{image_name.shellescape} #{url.shellescape}' download_#{image_name.shellescape}")
```

Comment:
there's some subtly weird stuff about this:

`common/bin/daemonizer 'host/bin/download-boot-image #{image_name.shellescape} #{url.shellescape}' download_#{image_name.shellescape}`

One is, nesting shellescapes in a quote. I mean, I think it will work, but it's a bit...odd I feel.

Probably the issue is daemonizer accepting a single string for the (presumed shell) program it accepts, rather than being capable of accepting argv with the unit name as something of an option:

`common/bin/daemonizer --unit=foo -- argv`

But that's not a problem for this patch.

`download_#{foo.shellescape}` is also a bit weird but probably works given the particular rules of shell, normally I'd expect one argv to be one `.shellescape` as the final operation, rather than mixing escaped and unescaped text, but it's something of a layering violation. But...maybe a profitable one for brevity?

The alternative would be:

`#{('download_' + image_name).shellescape}`

Reply:
Changed `download_#{foo.shellescape}` part. Yes, the first issue is a more general one, the codebase has similar usages.